python/cudf/benchmarks/common/utils.py

```diff
@@ -0,0 +1,267 @@
+
+    col_str = ""
+    if cols is not None:
+        assert cols in NUM_COLS, (
+            f"You have requested a DataFrame with {cols} columns but fixtures "
+            f"only exist for the values {', '.join(c for c in NUM_COLS)}"
+        )
+        col_str = f"_cols_{cols}"
+
+    row_str = ""
+    if rows is not None:
+        assert rows in NUM_ROWS, (
+            f"You have requested a {cls} with {rows} rows but fixtures "
+            f"only exist for the values {', '.join(c for c in NUM_ROWS)}"
+        )
+        row_str = f"_rows_{rows}"
+
+    fixture_name = f"{cls}{dtype_str}{null_str}{col_str}{row_str}"
+
+    def deco(bm):
+        # pytests's test collection process relies on parsing the globals dict
+        # to find test functions and identify their parameters for the purpose
+        # of fixtures and parameters. Therefore, the primary purpose of this
+        # decorator is to define a new benchmark function with a signature
+        # identical to that of the decorated benchmark except with the user's
+        # fixture name replaced by the true fixture name based on the arguments
+        # to accepts_cudf_fixture.
+        parameters = inspect.signature(bm).parameters
+
+        # Note: This logic assumes that any benchmark using this fixture has at
+        # least two parameters since they must be using both the
+        # pytest-benchmark `benchmark` fixture and the cudf object.
+        params_str = ", ".join(f"{p}" for p in parameters if p != name)
+        arg_str = ", ".join(f"{p}={p}" for p in parameters if p != name)
+
+        if params_str:
+            params_str += ", "
+        if arg_str:
+            arg_str += ", "
+
+        params_str += f"{fixture_name}"
+        arg_str += f"{name}={fixture_name}"
+
+        src = textwrap.dedent(
+            f"""
+            import makefun
+            @makefun.wraps(
+                bm,
+                remove_args=("{name}",),
+                prepend_args=("{fixture_name}",)
+            )
+            def wrapped_bm({params_str}):
+                return bm({arg_str})
+            """
+        )
+        globals_ = {"bm": bm}
+        exec(src, globals_)
+        wrapped_bm = globals_["wrapped_bm"]
+        # In case marks were applied to the original benchmark, copy them over.
+        if marks := getattr(bm, "pytestmark", None):
+            wrapped_bm.pytestmark = marks
+        wrapped_bm.place_as = bm
+        return wrapped_bm
+
+    return deco
+
+
+class OrderedSet(MutableSet):
+    """A minimal OrderedSet implementation built on a dict.
+
+    This implementation exploits the fact that dicts are ordered as of Python
+    3.7. It is not intended to be performant, so only the minimal set of
+    methods are implemented. We need this class to ensure that fixture names
+    are constructed deterministically, otherwise pytest-xdist will complain if
+    different threads have seemingly different tests.
+    """
+
+    def __init__(self, args=None):
```

Comment:
Why not `*values`? Make it act more like `set`.
```suggestion
    def __init__(self, *values):
```

Reply:
That isn't how set works
```
>>> set(1, 2, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: set expected at most 1 argument, got 3
>>> set((1, 2, 3))
{1, 2, 3}
>>> set([1, 2, 3])
{1, 2, 3}
```

Reply:
🤦‍♂️ Sorry, my bad.